### DISCUSSION TITLE
Replace generic lodash deps with more specific `lodash.x` deps

### DISCUSSION THREAD

**Discussion by martinheidegger** [babel/babel#12004] [General] [answered]
Trying to run a recent expo with babel project shows quite a few audit errors related to a security issue in lodash.

Example: https://www.npmjs.com/advisories/1523 affects `zipObjectDeep` which is not used at any place in the babel code from what I can tell.

Now I am wondering if it would be a good idea to replace all the unspecific `lodash` dependencies with specific ones. Like:

https://github.com/babel/babel/blob/eea156b2cb8deecfcf82d52aa1b71ba4995c7d68/packages/babel-types/package.json#L20

to be replaced with:

```json
  "lodash.isplainobject": "^4.0.6",
  "lodash.isregexp": "^4.0.1"
```

and subsequently replace

https://github.com/babel/babel/blob/eea156b2cb8deecfcf82d52aa1b71ba4995c7d68/packages/babel-types/src/converters/valueToNode.js#L2-L3

with

```javascript
 import isPlainObject from "lodash.isplainobject"; 
 import isRegExp from "lodash.isregexp"; 
```

This way any security audit issues that come up only show if they are relevant and likely the amount of packages loaded should be reduced.

Would it be a good idea to send PR's that replace all this?

**Comment by TrySound** [comment, marked as answer]
Was already raised. `lodash.*` packages are deprecated and gonna be removed in lodash v5.
https://github.com/babel/babel/pulls?q=is%3Apr+is%3Aopen+lodash

**Reply by martinheidegger** [reply]
Oh Cool! I thought I looked for it properly, but its awesome to hear.

**Comment by existentialism** [comment]
@martinheidegger also, FWIW, the package.json you linked to from babel-types is actually out of date: https://github.com/babel/babel/blob/main/packages/babel-types/package.json#L20

**Reply by martinheidegger** [reply]
Thanks for pointing that out. I was really just trying to see what the sentiment is around the topic and if I can help but you are hard at work on it, and I am sorry to have disturbed you. Thanks for the answers and the good work!

**Reply by existentialism** [reply]
@martinheidegger there is absolutely no need to apologize! we appreciate you looking into this!
